CIP-0030/README.md

```diff
@@ -194,15 +234,60 @@ In order to initiate communication from webpages to a user's Cardano wallet, the
+DApps are expected to use this endpoint to perform an initial handshake and ensure that the wallet supports all its required functionalities.
+
+*Note:* Extensions can be mutually incompatible.
+Incompatibilities can arise for example:
+
+- When extensions provide conflicting features.
+- If a newer extension wholly replaces the functionality of an older one.
+- A draft `"x-cip"` is replaced by a final `"cip"`.
+
+While conflicting extensions should be avoided where reasonable while designing CIPs, it is also the responsibility of wallet providers to assess whether they can support a given combination of extensions, or not.
+Wallets *MUST NOT* fail to enable the API should they not recognize or not support a particular combination of extensions.
+Instead, they should decide what they enable and reflect their choice in the [`cardano.{walletName}.extensions`](#cardanowalletnamesupportedextensions-extension) property of the Full API.
+DApps will use the extensions reported by that property to decide how to proceed.
+For example, they can fail and inform their users that necessary wallet features are unavailable, or may use a different, less-efficient, strategy to cope with a lack of functionality.
+This may require the wallet to re-request a different set of extensions.
 
 ##### Can extensions depend on other extensions?
 
-Yes. Extensions may have other extensions as pre-requisite. Some newer extensions may also invalidate functionality introduced by earlier extensions. There's no particular rule or constraints in that regards. Extensions are specified as CIP, and will define what it entails to enable them.
+Yes. Extensions may have other extensions as pre-requisite. Some newer extensions may also invalidate functionality introduced by earlier extensions. There are no particular rules or constraints in that regard. Extensions are specified as a CIP, and will define what it entails to enable them.
+
+Wallets should attempt to enable all extensions necessary to enable the extensions requested, even if they were not specifically requested.
+For example, if `CIP-9999` needs `CIP-8123` and a request is made to ONLY enable `[{"cip":9999}]` then the wallet should automatically enable `CIP-8123` as if the request was actually for `[{"cip":8123},{"cip":9999}]`.
+
+##### Can wallets enable more extensions than requested?
+
+Yes.
+It would be a legitimate implementation of this CIP for a wallet to **ALWAYS** enable every extension it supports, regardless of the requested list of extensions.
+The only caveat is that the [`cardano.{walletName}.extensions`](#cardanowalletnamesupportedextensions-extension) property must faithfully reflect the extensions which the wallet API object supports.
 
 ##### Should extensions follow a specific format?
```

Comment:
Prospective extension CIPs should clearly outline and log their respective "x-cip" numbering/naming.

Reply:
Agree, wallets should list all extensions they support,  both experimental/draft and finalized CIPs.